src/xdp-utils.c

```diff
@@ -317,3 +317,71 @@ xdg_desktop_portal_error_quark (void)
   return (GQuark) quark_volatile;
 }
 
+char *
+xdp_get_path_for_fd (GKeyFile *app_info,
+                     int fd)
+{
+  g_autofree char *proc_path = NULL;
+  int fd_flags;
+  char path_buffer[PATH_MAX + 1];
+  g_autofree char *rewritten_path = NULL;
+  char *path;
+  ssize_t symlink_size;
```

Comment:
I know this was not a problem before, but I think it would be nice to initialize this to zero. It will be used to do `path_buffer[symlink_size] = 0` and not being assigned any value at that point is a possibility, so I think I'd rather not rely on undefined behaviour here.